cmd/soroban-cli/src/signer/mod.rs

```diff
@@ -326,31 +332,3 @@ impl Lab {
         Err(Error::ReturningSignatureFromLab)
     }
 }
-
-pub struct SecureStoreEntry {
-    pub name: String,
-    pub hd_path: Option<usize>,
-}
-
-impl SecureStoreEntry {
```

Comment:
I moved this over to secure_store_entry.rs for 2 reasons:
1. so we could merge this `SecureStoreEntry` functionality with what was in `secure_store.rs` - this just removed a layer that we didnt really need, and allows SecureStoreEntry access the keyring entry directly 
2. we want to feature flag some of the imports required for the keyring crate so that users can build the binary without the additional-libs feature and it seemed better to do that in a separate file